core/templates/dev/head/services/contextual/UrlServiceSpec.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the BackgroundMaskService.
+ */
+
+describe('Url Service', function() {
+  var UrlService = null;
+  var parameterList = '?parent=parent1&parent=parent2';
+  var sampleHash = 'sampleHash';
+  var window = {
+    href: 'http://sample.com/embed' + parameterList,
+    pathname: 'sample.com/embed',
+    hash: sampleHash
+  };
+
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    UrlService = $injector.get('UrlService');
+    spyOn(UrlService, 'getCurrentUrl').and.returnValue(window);
+  }));
+
+  it('should correctly get last url parameter', function() {
```

Comment:
getUrlParams() doesn't sound like "last url parameter" to me.

Also, this should check for cases with multiple params, no params, multiple field values with the same field name, mixtures of these, etc. Please test thoroughly since we're going to be relying on the behaviour of this service in the future.

Reply:
Yes, I thought of that too, but this was a function that was already present and getQueryFieldValuesAsList() does almost the same function, so shall I remove getUrlParams()?

Reply:
I think it's fine to keep both as separate functions, actually. It depends on whether other places use getUrlParams(). If so, you can keep both (just make sure they work properly in all cases) and, if not, you can ditch getUrlParams().

Reply:
Actually UrlParams(), returns all field name - value pair in an object form, so I am keeping it.
Also, if a fieldname occurs more than once, it only returns last value, for that purpose (to get in array form) getQueryFieldValuesAsList() can be used. Wrote a comment regarding this.